DESIGN.md

```diff
@@ -0,0 +1,173 @@
+# Design
+
+## User experience
+
+TLS certificates are expected to be generated out-of-band ahead of time.
+
+### Starting the server
+
+Options:
+
+* `-c`/`--cert`: takes a path to the server's TLS certificate.
+* `-k`/`--key`: takes a path to the server's TLS key.
+* `-a`/`--address`: takes an address to listen on.
+* `-p`/`--port`: takes a port number to listen on.
+* `-r`/`--remote`: takes a pair of permission level and client certificate path.
+  May be specified multiple times.
```

Comment:
In tandem with the thread at https://github.com/CobaltCause/go-remote/pull/1/files#r2298415855, I wonder if the UX here could be improved by not having to specify the permission level? Maybe (if necessary) the client could infer the permission level from the cert on its own? It's no showstopper, but CLI flags that accept two arguments are a little bit unusual.